src/lib/comms/sol-lwm2m.c

```diff
@@ -3759,9 +4158,80 @@ lifetime_client_timeout(void *data)
 }
 
 SOL_API int
+sol_lwm2m_bootstrap_server_send_finish(struct sol_lwm2m_bootstrap_server *server,
+    struct sol_lwm2m_bootstrap_client_info *client)
+{
+    struct sol_coap_packet *pkt;
+    int r;
+
+    SOL_NULL_CHECK(server, -EINVAL);
+    SOL_NULL_CHECK(client, -EINVAL);
+
+    pkt = sol_coap_packet_new_request(SOL_COAP_METHOD_POST, SOL_COAP_MESSAGE_TYPE_CON);
+    r = -ENOMEM;
+    SOL_NULL_CHECK_GOTO(pkt, err_exit);
+
+    r = sol_coap_add_option(pkt, SOL_COAP_OPTION_URI_PATH, "bs", strlen("bs"));
+    SOL_INT_CHECK_GOTO(r, < 0, err_coap);
+
+    SOL_DBG("Sending Bootstrap Finish to LWM2M Client %s", client->name);
+    r = sol_coap_send_packet(server->coap, pkt, &client->cliaddr);
+    return r;
```

Comment:
You're leaking the client info. You should delete it.
We should also mention in the docs that after `sol_lwm2m_bootstrap_server_send_finish()` returns the client_info handle will be INVALID!